### ISSUE TITLE
RRULE UNTIL Property Should Be Inclusive

### ISSUE THREAD

**Issue by ge1157** [ical-org/ical.net#320]
In the case below the getOccurences() method is not returning the last day of the repeated event.

```cs
        [TestMethod]
        public void FiveDayRRule()
        {
            const string icalText = @"BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTART;VALUE=DATE:20180101
DTEND;VALUE=DATE:20180107
RRULE:FREQ=WEEKLY;UNTIL=20180105;BYDAY=MO,TU,WE,TH,FR
DTSTAMP:20170926T001103Z
UID:5kvks79u4nurqopt7qv4fi1jo8@google.com
CREATED:20170922T131958Z
DESCRIPTION:
LAST-MODIFIED:20170922T131958Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Holiday Break - No School
TRANSP:TRANSPARENT
END:VEVENT
END:VCALENDAR
";
            const string timeZoneId = @"Eastern Standard Time";
            CalendarCollection collection = Calendar.LoadFromStream(new StringReader(icalText));
            var firstEvent = collection.First().Events.First();
            var startSearch = new CalDateTime(DateTime.Parse("2017-07-01T00:00:00"), timeZoneId);
            var endSearch = new CalDateTime(DateTime.Parse("2018-07-01T00:00:00"), timeZoneId);

            var occurrences = firstEvent.GetOccurrences(startSearch, endSearch)
                .Select(o => o.Period as Period)
                .OrderBy(p => p.StartTime)
                .ToList();

            Console.WriteLine("FiveDayRRule() found {0} occurences.", occurrences.Count());
            Assert.IsTrue(occurrences.Count() == 5);
        }
```

**Comment by rianjs** [comment]
Your DTEND makes no sense. You’ve specified a 7 day long event that repeats daily for five days.

**Comment by ge1157** [comment]
My apologies for the offending DTEND. Following is the code using the original VEVENT snippet:
````
[TestMethod]
        public void FiveDayRRule()
        {
            const string icalText = @"BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTART;VALUE=DATE:20180101
DTEND;VALUE=DATE:20180102
RRULE:FREQ=WEEKLY;UNTIL=20180105;BYDAY=MO,TU,WE,TH,FR
DTSTAMP:20170926T001103Z
UID:5kvks79u4nurqopt7qv4fi1jo8@google.com
CREATED:20170922T131958Z
DESCRIPTION:
LAST-MODIFIED:20170922T131958Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Holiday Break - No School
TRANSP:TRANSPARENT
END:VEVENT
END:VCALENDAR
";
            const string timeZoneId = @"Eastern Standard Time";
            CalendarCollection collection = Calendar.LoadFromStream(new StringReader(icalText));
            var firstEvent = collection.First().Events.First();
            var startSearch = new CalDateTime(DateTime.Parse("2017-07-01T00:00:00"), timeZoneId);
            var endSearch = new CalDateTime(DateTime.Parse("2018-07-01T00:00:00"), timeZoneId);

            var occurrences = firstEvent.GetOccurrences(startSearch, endSearch)
                .Select(o => o.Period as Period)
                .OrderBy(p => p.StartTime)
                .ToList();

            Console.WriteLine("FiveDayRRule() found {0} occurences.", occurrences.Count());
            foreach (var entry in occurrences)
            {
                Console.WriteLine("{0} to {1}", entry.StartTime, entry.EndTime);
            }
            Assert.IsTrue(occurrences.Count() == 5);
        }
````
It only produces 4 occurrences. Using Ical.Net v3.0.12 obtained in VS2013 using NuGet.

**Comment by rianjs** [comment]
Fixed in nuget version 4.0.0: https://www.nuget.org/packages/Ical.Net/4.0.0
